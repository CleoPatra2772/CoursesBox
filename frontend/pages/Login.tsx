import type { NextPage } from "next";
import Link from "next/link";
import styled from "@emotion/styled";
import { useForm } from "react-hook-form";

import { CenteredTile } from "../component/Tile";
import { Input} from "@/component/Input";
import { Feedback } from "@/component/Input/Feedback";
import { Button } from "@/component/Button";
import { StyledLink } from "@/component/StyledLink";

const StyledInput = styled(Input)`
margin-bottom: 1rem;
`;

export type LoginForm = {
    identifier: string;
    password: string;
};


const Login: NextPage = () => {
    const { register, handleSubmit, formState: { errors } } = useForm<LoginForm>();


    const onSubmit = (data: LoginForm) => {
        console.log(data);
    };
    return (
   
    <form onSubmit={handleSubmit(onSubmit)}>
         <CenteredTile header="Login">
        <StyledInput 
        label="Identifier" 
        placeholder="username or email"
        feedback = {
            errors.identifier ? <Feedback>Min Length 6!</Feedback> : <>&nbsp;</>}
        height={8}
        {...register("identifier", {
            required: true,
            minLength: 6,
        })}
        />
        <StyledInput 
        label="Password" 
        type="password" 
        placeholder="password"
        feedback = {
            errors.password ? <Feedback>Min Length 6!</Feedback> : <>&nbsp;</>}
        height={8} 
        {...register("password", {
            required: true,
            minLength: 6,
        })}
        />
        <Button type="submit">Sign In</Button>
        <h3>
            <Link href="/registration" passHref>
                <StyledLink underline>Create account</StyledLink>
            </Link>
        </h3>
        </CenteredTile>
    </form>
    
    );
};

export default Login;