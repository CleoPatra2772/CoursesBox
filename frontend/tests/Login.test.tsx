import { render } from "@/test-utils";
import Login from "@/pages/Login";

describe("Login page", () => {

    it("render check", () =>{
        const { container } = render(<Login />);
        expect(container).toMatchSnapshot();
    });
});