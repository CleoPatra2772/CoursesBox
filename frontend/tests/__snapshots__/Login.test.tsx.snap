// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login page render check 1`] = `
<div>
  <form>
    <div
      class="css-1owwovu"
    >
      <section
        class="css-1fluu6k"
      >
        <h2>
          login
        </h2>
         
        <label
          class="css-1xs7g1t"
          height="8"
          width="20"
        >
          <span
            class="css-6hin7w"
          >
            Identifier
          </span>
          <div
            class="css-qqgi3g"
          >
            <input
              class="css-1ma8s4p"
              name="identifier"
              placeholder="username or email"
              role="textbox"
            />
          </div>
          <span
            class="css-8t7t1n"
            role="alert"
          >
             
          </span>
        </label>
        <label
          class="css-1xs7g1t"
          height="8"
          width="20"
        >
          <span
            class="css-6hin7w"
          >
            Password
          </span>
          <div
            class="css-qqgi3g"
          >
            <input
              class="css-1ma8s4p"
              name="password"
              placeholder="password"
              role="textbox"
              type="password"
            />
          </div>
          <span
            class="css-8t7t1n"
            role="alert"
          >
             
          </span>
        </label>
        <button
          class="css-tjwez1"
          color="primary"
          type="submit"
        >
          Sign In
        </button>
        <h3>
          <a
            class="css-1ixycxi"
            href="/registration"
          >
            Create account
          </a>
        </h3>
         
      </section>
    </div>
  </form>
</div>
`;
